numpy/core/src/umath/reduction.c

```diff
@@ -493,6 +484,13 @@ PyUFunc_ReduceWrapper(PyArrayObject *operand, PyArrayObject *out,
         Py_INCREF(op_view);
     }
     else {
+        /* Cannot use where when we initialize from the operand */
+        if (wheremask != NULL) {
+            PyErr_SetString(PyExc_RuntimeError,
+                            "Reduce operations with no identity only support "
+                            "a where mask if 'initial' is passed in.");
```

Comment:
Might be nice to include the ufunc name in this message.

Reply:
Yes. Actually, what error do you think this should be? `RunTimeError` is more for something that is unexpected, where this is not. Just `ValueError`? I guess I should catch it earlier, in the keyword getting stage (can keep it here just in case, though `ReduceWrapper` is not part of the C API so it is just for internal checking).